commands/command_checkout.go

```diff
@@ -35,12 +35,10 @@ func checkoutCommand(cmd *cobra.Command, args []string) {
 	close(inchan)
 
 	filter := filepathfilter.New(rootedpaths, nil)
-	gitscanner := lfs.NewGitScanner(nil)
-	defer gitscanner.Close()
-	checkoutWithIncludeExclude(gitscanner, filter)
+	checkoutWithIncludeExclude(filter)
 }
 
-func checkoutFromFetchChan(gitscanner *lfs.GitScanner, filter *filepathfilter.Filter, in chan *lfs.WrappedPointer) {
+func checkoutFromFetchChan(in chan *lfs.WrappedPointer) {
```

Comment:
I think this function still needs a `*filepathfilter.Filter`. Additionally, we should probably put a safety net around this and guard it with an integration test.